core/cas-server-core-events/src/test/java/org/apereo/cas/support/events/CasAuthenticationEventListenerTests.java

```diff
@@ -135,20 +160,94 @@ public void verifyCasRiskyAuthenticationDetectedEvent() {
             CoreAuthenticationTestUtils.getAuthentication(),
             CoreAuthenticationTestUtils.getRegisteredService(),
             new Object());
+        var current = numEventsReceived.get();
         applicationContext.publishEvent(event);
+        sleep(current + 1);
         assertFalse(casEventRepository.load().findAny().isEmpty());
     }
 
     @Test
     public void verifyCasTicketGrantingTicketDestroyed() {
         val event = new CasTicketGrantingTicketDestroyedEvent(this,
             new MockTicketGrantingTicket("casuser"));
+        var current = numEventsReceived.get();
         applicationContext.publishEvent(event);
+        sleep(current + 1);
         assertFalse(casEventRepository.load().findAny().isEmpty());
     }
+    @Test
+    public void verifyEventRepositoryHasOneEventOnly() {
+        clearEventRepository();
+        val event = new CasTicketGrantingTicketDestroyedEvent(this,
+                new MockTicketGrantingTicket("casuser"));
+        var current = numEventsReceived.get();
+        applicationContext.publishEvent(event);
+        sleep(current + 1);
+        assertEquals(1,casEventRepository.load().count());
+    }
+
+    @Test
+    public void verifyCasTicketGrantingTicketDestroyedHasClientInfo() {
+        clearEventRepository();
+        val event = new CasTicketGrantingTicketDestroyedEvent(this,
+                new MockTicketGrantingTicket("casuser"));
+        var current = numEventsReceived.get();
+        applicationContext.publishEvent(event);
+        sleep(current + 1);
+        val result = casEventRepository.load().toList().get(0).getClientIpAddress();
```

Comment:
You should assert that the list exists and contains elements first, or else the test fails with a completely different error, which is often not helpful even though accurtate.

Reply:
I added an assert to check to see if the repository has values before the toList() call.